test/test_build.py

```diff
@@ -1,79 +1,139 @@
+    def test_build_path_default(self, language, command):
+        runner = CliRunner()
+        project_path = self.python_project_path if language == "python" else self.scala_project_path
+        result = runner.invoke(command, ["--path", project_path])
+        print(result.stdout)
+        assert result.exit_code == 0
+        assert "Found 5 pipelines" in result.output
+        TestBuild._check_for_artifacts(project_path, language, n=5)
 
-def test_build_v2_path_default_build_errors():
-    runner = CliRunner()
-    result = runner.invoke(build_v2, ["--path", ERROR_PROJECT_PATH])
-    assert result.exit_code == 1
+    @pytest.mark.parametrize(
+        "language, build_file_name",
+        [
+            ("python", "setup.py"),
+            ("scala", "pom.xml"),
+        ],
+    )
+    def test_build_v2_path_default_build_errors(self, language, build_file_name):
+        project_path = self.python_project_path if language == "python" else self.scala_project_path
+        a_file_to_mangle = list(
+            glob.glob(os.path.join(project_path, "pipelines", "**", build_file_name), recursive=True)
+        )[0]
+        print("file to mangle: ", a_file_to_mangle)
+        with open(a_file_to_mangle, "w") as fd:
+            fd.write("oops")
+            fd.close()
+        runner = CliRunner()
+        result = runner.invoke(build_v2, ["--path", project_path])
+        print(result.stdout)
+        assert result.exit_code == 1
+        TestBuild._check_for_artifacts(project_path, language, n=4)
 
+    @pytest.mark.parametrize(
+        "language, build_file_name",
+        [
+            ("python", "setup.py"),
+            ("scala", "pom.xml"),
+        ],
+    )
+    def test_build_v2_path_default_build_errors_ignore_errors(self, language, build_file_name):
+        project_path = self.python_project_path if language == "python" else self.scala_project_path
+        a_file_to_mangle = list(
+            glob.glob(os.path.join(project_path, "pipelines", "**", build_file_name), recursive=True)
+        )[0]
+        print("file to mangle: ", a_file_to_mangle)
+        with open(a_file_to_mangle, "w") as fd:
+            fd.write("oops")
+            fd.close()
+        runner = CliRunner()
+        result = runner.invoke(build_v2, ["--path", project_path, "--ignore-build-errors"])
+        print(result.stdout)
+        assert result.exit_code == 0
+        TestBuild._check_for_artifacts(project_path, language, n=4)
 
-def test_build_v2_path_default_build_errors_ignore_errors():
-    runner = CliRunner()
-    result = runner.invoke(build_v2, ["--path", ERROR_PROJECT_PATH, "--ignore-build-errors"])
-    assert result.exit_code == 0
+    @pytest.mark.parametrize("language", ["python", "scala"])
+    @pytest.mark.parametrize("command", [build, build_v2])
+    def test_build_path_pipeline_filter(self, language, command):
+        project_path = self.python_project_path if language == "python" else self.scala_project_path
```

Comment:
minor: maybe we can make this a method in `IsolatedRepoTestCase`

Reply:
good idea, will do